drivers/gpio/gpio_nrf5.c

```diff
@@ -375,6 +383,28 @@ static const struct gpio_driver_api gpio_nrf5_drv_api_funcs = {
 	.disable_callback = gpio_nrf5_disable_callback,
 };
 
+/* Enable GPIOTE Interrupt */
+void nrf_gpiote_interrupt_enable(uint32_t mask)
+{
+	NRF_GPIOTE->EVENTS_PORT = 0;
+	nrf_gpiote_int_enable(mask);
+}
+
+/* Disable GPIOTE Interrupt */
+void  nrf_gpiote_interrupt_disable(uint32_t mask)
+{
+	NRF_GPIOTE->EVENTS_PORT = 0;
+	nrf_gpiote_int_disable(mask);
+}
+
+/* Clear GPIOTE Port Event */
+void nrf_gpiote_clear_port_event(void)
+{
+	if (NRF_GPIOTE->EVENTS_PORT) {
```

Comment:
its optimal to just set to 0 (always), than have a branch operations (if-clause). Check if you can reuse code to clear EVENT, i.e. any nrf_gpiote_* function that clears EVENT.

Reply:
Good point, we can remove the condition. I had put it as it was put in nrf sample application. But i agree we can remove it.

Also i do not see any reference to function in HAL to clear GPIOTE Event.

Reply:
The sample is managing LED. But in this PR, the if-clause is not needed, please remove.